documentation/swagger-checklist.md

```diff
@@ -43,82 +45,236 @@ This document defines a Swagger checklist for Microsoft Azure, along with clarif
+
+- [ ] **M2026**: If a property or a parameter has an `"enum": []` constraint and it also has a `"default"` value then the default value *MUST* be one of the values specified in the enum. It is *RECOMMENDED* that the default value be the first element in the "enum" array.
+
+- [ ] **M2027**: The value of [`x-ms-client-name` extension](https://github.com/Azure/autorest/blob/master/Documentation/swagger-extensions.md#x-ms-client-name) *MUST NOT* be an empty string, because it will be used as the identifier of the artifact in the generated client.
+
+- [ ] **M2028**: Every path in the `"paths": { . . . }` object *MUST* be unqiue and *MUST NOT* be equivalent. 
+  - `/paths/path1/{param1}` and `/paths/path1/{param2}` are equivalent paths and are *NOT* allowed.
+
+- [ ] **M2029**: Properties that are marked as `"required"` *MUST* exist in the `properties: { }` object of the model definition ([Schema Object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#schemaObject)) or one of its ancestors.
+
+- [ ] **M2030**: Every model defined in the definitions section of the swagger specification *MUST* have a `"type"` property. The list of supported data types in 2.0 swagger specification can be found [here](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#data-types).
+
+- [ ] **M2031**: Operations *MUST NOT* have both a `body` parameter and a `formData` parameter.
+
+- [ ] **M2032**: Operations *MUST* have **exactly one** `body` parameter.
+
+- [ ] **M2033**: Operations *MUST* have unique (name + in combination) parameters.
+
+- [ ] **M2034**: Operations *MUST* have a unique operationId in the swagger specification.
+
+- [ ] **M2035**: Path parameters declared in the path string *MUST* have matching parameter definitions (either at the path-level or the operation) and vice-versa (extra defintions or declarations are *NOT* allowed).
+
+- [ ] **M2036**: Path parameter declarations *MUST NOT* allow empty names (/path/{} is not valid).
+
+- [ ] **M2037**: Paths *MUST* have unique (name + in combination) parameters.
+
+- [ ] **M2038**: Any local (model or parameter) reference *MUST* start with `"#"`. Example `"#/parameters/subscriptionId"` or `"#/definitions/Resource"`.
+
+- [ ] **M2039**: Any external (model or parameter) reference, whether a relative or absolute file or url reference *MUST* adhere to the [JSON pointer reference syntax](https://tools.ietf.org/html/rfc6901). `"<(absolute|relative) (file|url) path>#<(definitions|parameters|or any other key that needs to be accessed in the json object at that location)"`. The pointer reference traverses ahead with a forward slash. Example: `"$ref": "./spec2.json#/definitions/Model1"`.
+
+- [ ] **M2040**: Circular composition/inheritance for Schema Objects *MUST NOT* be presnt. A spec can have circular references everywhere except in composition/inheritance.
```

Comment:
M00000 - No spelling mistakes should be present in swagger review list :) [Presnt]